unison.opam

```diff
@@ -11,10 +11,9 @@ bug-reports: "https://github.com/bcpierce00/unison/issues"
 dev-repo: "git://github.com/bcpierce00/unison.git"
 build: ["dune" "build" "-p" name "-j" jobs]
 depends: [
-  "ocaml" {build & >= "4.08"}
-  "ocamlfind" {build}
-  "dune" {build & >= "2.3"}
```

Comment:
Can you explain why dune would be needed at runtime?  (or why I'm confused?)

Reply:
Because of https://github.com/ocaml/dune/issues/2147 dune is not backward compatible and would cause problem when downgrading.

Reply:
Why is dune involved after the package is built?   Is this about a dune metadata file that is interpreted if some other package depends on unison?  (Are there any such packages?)   Or because people expect to invoke some dune commands on installed opam packages in general, or unison in particular?

Reply:
This one I still don't understand.

It seems like some people are telling me that opam is primary and others are telling me that dune-project should generate unison.opam, and my read is that there are differences of opinion within the ocaml community.  So I'm trying extra hard to understand.

Reply:
Dune installs a `dune-package` file that have the version of dune it has been compiled with:
```
$ cat ~/.opam/default/lib/unison/dune-package
(lang dune 3.5)
(name unison)
(version dev)
(sections (lib .) (bin ../../bin) (doc ../../doc/unison))
(files
 (lib (META dune-package opam))
 (bin (unison))
 (doc (LICENSE README.md)))
```